### PR TITLE
chore(provider): publish provider

### PR DESCRIPTION
**Pull request by azman0101**
Fix https://github.com/claranet/terraform-provider-zabbix/issues/18
The secret GPG_PRIVATE_KEY and PASSPHRASE must be added in secrets to allow gorelease to sign the providers.
https://github.com/claranet/terraform-provider-zabbix/settings/secrets

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Go 1.13
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.13.x
```
Comment (pdecat) [inline .github/workflows/release.yml:27]: I'd rather use the same version for test and for [release](https://github.com/claranet/terraform-provider-zabbix/pull/19/files#diff-56c30d9224897b08c28075052d4595c5R45).

Why not go 1.15.2?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# You will need to pass the `--batch` flag to `gpg` in your signing step
+# in `goreleaser` to indicate this is being used in a non-interactive mode.
+#
+name: release
+on:
+  push:
+    tags:
+      - 'v*'
+jobs:
+  test:
```
Comment (pdecat) [inline .github/workflows/release.yml:18]: The `test` job should still run on all `pull_request` and `push` events as before: https://github.com/claranet/terraform-provider-zabbix/pull/19/files#diff-c2e34dca95d7bdcdd2e85f92915cdda8L3-L5

Then the `if` condition from the old workflow should be added to the `release`: https://github.com/claranet/terraform-provider-zabbix/pull/19/files#diff-c2e34dca95d7bdcdd2e85f92915cdda8L25

That's why the workflow was named `Test and Release`.

Or maybe we could separate the test job from the release job in another workflow.

WDYT?

File: .github/workflows/release.yml (hunk 1 above)
Comment (oxalide-mp-tech) [reply .github/workflows/release.yml:27]: Why not, I've just cut&paste [the boilerplate](https://github.com/hashicorp/terraform-provider-scaffolding/blob/master/.goreleaser.yml) referenced in https://www.terraform.io/docs/registry/providers/publishing.html#publishing-to-the-registry

Comment (azman0101) [reply .github/workflows/release.yml]: fixed

File: .github/workflows/release.yml (hunk 2 above)
Comment (azman0101) [reply .github/workflows/release.yml:18]: fixed

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        run: git fetch --prune --unshallow
+      -
+        name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15.3
+      -
+        name: Import GPG key
+        id: import_gpg
+        uses: paultyng/ghaction-import-gpg@v3.0.1
```
Comment (pdecat) [inline .github/workflows/release.yml:50]: That fork does not have the v3.0.1 tag.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        run: git fetch --prune --unshallow
+      -
+        name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15.3
+      -
+        name: Import GPG key
+        id: import_gpg
+        uses: crazy-max/ghaction-import-gpg@v3.0.1
```
Comment (pdecat) [inline .github/workflows/release.yml:50]: FTR, this is also the action recommended by goreleaser: https://goreleaser.com/ci/actions/#signing

**Review by pdecat — APPROVED** [review]
LGTM!

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# This GitHub action can publish assets for release when a tag is created.
+# Currently its setup to run on any tag that matches the pattern "v*" (ie. v0.1.0).
```
Comment (pdecat) [inline .github/workflows/release.yml:2]:
```suggestion
# Currently it's setup to run on any tag that matches the pattern "v*" (ie. v0.1.0).
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+#
+# This uses an action (paultyng/ghaction-import-gpg) that assumes you set your
```
Comment (pdecat) [inline .github/workflows/release.yml:4]:
```suggestion
# This uses an action (crazy-max/ghaction-import-gpg) that assumes you set your
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15.3
+      -
+        name: Import GPG key
+        id: import_gpg
+        uses: crazy-max/ghaction-import-gpg@v3.0.1
+        env:
+          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+          PASSPHRASE: ${{ secrets.PASSPHRASE }}
```
Comment (pdecat) [inline .github/workflows/release.yml:53]:
```suggestion
        with:
          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
          passphrase: ${{ secrets.PASSPHRASE }}
```

**Review by pdecat — CHANGES_REQUESTED** [review]
Last minute changes following switch to crazy-max/ghaction-import-gpg@v3.0.1.

File: .github/workflows/release.yml (hunk 4 above)
Comment (azman0101) [reply .github/workflows/release.yml:50]: 👍

**Review by pdecat — APPROVED** [review]

**Comment by barbaluc** [comment]
Hi !

Dispite the merged pull request, the provider is still missing from the terraform registry : https://registry.terraform.io/search/providers?q=zabbix
Can you take a look at it ?

**Comment by pdecat** [comment]
The provider is now showing in the search results but https://registry.terraform.io/providers/claranet/zabbix/latest is still in 404.

**Comment by pdecat** [comment]
And now it's ok: https://registry.terraform.io/providers/claranet/zabbix/latest

**Comment by barbaluc** [comment]
Thanks @pdecat, it works as expected !

**Pull request merged**